openhtmltopdf-core/src/main/java/com/openhtmltopdf/extend/Hyphenator.java

```diff
@@ -0,0 +1,8 @@
+package com.openhtmltopdf.extend;
+
+/**
+ * Interface for a custom hyphenation implementation.
+ */
+public interface Hyphenator {
+    String hyphenateText(String text);
```

Comment:
At first glance, I'm not sure the single `String` parameter to this function provides sufficient information to properly transform the text. I would expect the following information to be relevant:

- Maximum width of the container
- Options specified by the `hyphenate-limit-chars` CSS property.

Reply:
For the hyphenator itself, I think a String as paramenter is sufficent. The call of the hyphenator could be optimised on the other hand, since it currently puts soft-hyphens at all defined spots of the entire text and further uses the already implemented soft-hyphens support. Ideally I would like to call the hyphenator only for words, that need to be split, but I couldn't find the appropriate spot yet.
Due to some time constraints within my current project, I was going for this approach, but I do want to optimize it in my free time.

Reply:
The soft hyphen approach is interesting. I hadn't considered that, but it makes sense. I agree with your assessment regarding optimization.